.env.example

```diff
@@ -14,6 +14,13 @@ DJANGO_ALLOWED_HOSTS="localhost 127.0.0.1 0.0.0.0 app nginx"
 
 SECRET_KEY=change_me
 
+# Certificates
+# For a local instance self signed certs are used.
+# For production use with certbot and letsencrypt uncomment the following lines:
+# QFIELDCLOUD_TLS_CERT=/etc/letsencrypt/live/${QFIELDCLOUD_HOST}/fullchain.pem
+# QFIELDCLOUD_TLS_KEY=/etc/letsencrypt/live/${QFIELDCLOUD_HOST}/privkey.pem
+# QFIELDCLOUD_TLS_DHPARAMS=/etc/nginx/dhparams/dhparams4096.pem
```

Comment:
For me the main question here was how to make dev environments working without any changes in the .env and minimal changes to a production setup. When pointing default to the letsencrypt here, manually changes would be needed here for dev environments. 

As you have to/should change variables in .env in a production setup, i would prefer it this way with uncomments for production. But i can change it to whatever you prefer.

Reply:
We are on the same page, I also prefer minimal changes. What I usually try to avoid is uncommeting stuff, to me it's never clear who, when and why commented/uncommented stuff. Isn't it going to work with minimal to no changes if we just do:

```
QFIELDCLOUD_TLS_CERT=""
QFIELDCLOUD_TLS_KEY=""
QFIELDCLOUD_TLS_DHPARAMS=""
```

And if one wants to have let's encrypt, add it in the comment documenting each value? Rough suggestion:

```

# TLS certificate filename from within the `nginx` container.
# When empty, the default self-signed certificate is used.
# For usage with Let's Encrypt certificate, use as:
# QFIELDCLOUD_TLS_CERT="/etc/letsencrypt/live/${QFIELDCLOUD_HOST}/fullchain.pem"
# DEFAULT: ""
QFIELDCLOUD_TLS_CERT=""

```

Reply:
```suggestion
# TLS certificate filename from within the `nginx` container.
# When empty, the default self-signed certificate is used.
# For usage with Let's Encrypt certificate, use as:
# QFIELDCLOUD_TLS_CERT="/etc/letsencrypt/live/${QFIELDCLOUD_HOST}/fullchain.pem"
# QFIELDCLOUD_TLS_KEY="/etc/letsencrypt/live/${QFIELDCLOUD_HOST}/privkey.pem"
# QFIELDCLOUD_TLS_DHPARAMS="/etc/nginx/dhparams/dhparams4096.pem"
# DEFAULT for all: ""
QFIELDCLOUD_TLS_CERT=""
QFIELDCLOUD_TLS_KEY=""
QFIELDCLOUD_TLS_DHPARAMS=""
```
Is it fine like this?

Reply:
```suggestion
# Certificates

# TLS certificate filename from within the `nginx` container.
# When empty, the default self-signed certificate is used.
# For usage with Let's Encrypt certificate, use as:
# QFIELDCLOUD_TLS_CERT="/etc/letsencrypt/live/${QFIELDCLOUD_HOST}/fullchain.pem"
# DEFAULT: ""
QFIELDCLOUD_TLS_CERT=""

# TLS private key filename from within the `nginx` container.
# When empty, the default self-signed certificate is used.
# For usage with Let's Encrypt certificate, use as:
# QFIELDCLOUD_TLS_KEY="/etc/letsencrypt/live/${QFIELDCLOUD_HOST}/privkey.pem"
# DEFAULT: ""
QFIELDCLOUD_TLS_KEY=""

# TLS DF parameters filename from within the `nginx` container.
# When empty, no DF parameters are used.
# DEFAULT: "/etc/nginx/dhparams/dhparams4096.pem"
QFIELDCLOUD_TLS_DHPARAMS="/etc/nginx/dhparams/dhparams4096.pem"
```

The DH params can be set by default to use the one included in the repo, no matter if we use self-signed or let's encrypt params?

Reply:
Thx.

About the DH Parameters:
- Using precomputed shared parameters increases the attack surface for precomputation attacks
- Using self generated increases the risk for weak primes. 
- Self generating consumes more Energy. 

Both has pros and cons and in the end it doesn't matter I think.

I will adapt the script later to precomputed.

Reply:
I now use the RFC 3526 4096bit DH params by default, other DH-params are ignored by git.

I have not tested it jet again.